### ISSUE TITLE
all notes and tags duplicated after app crash and encryption key retrieval

### ISSUE THREAD

**Issue by chaserene** [standardnotes/mobile#327]
3.5.8
Android 9
installed via APK from GitHub

I was trying to open a note when the app crashed. I opened the app again, this time I only had to enter my passcode. I tried opening the same note again, and it crashed again. (I have no idea why. This note uses Minimal Markdown, which has been working without issues on other notes.)

After the second crash, as I already described a similar case {on Slack](https://standardnotes.slack.com/archives/C3LC7FFP1/p1606239436098900?thread_ts=1606154089.080100&cid=C3LC7FFP1), I got the below request when reopening the app:

![image](https://user-images.githubusercontent.com/64873595/100392777-efd54c80-3037-11eb-909e-ac5050ce2c21.png)

after entering my password, things took a different turn. I was taken to my notes, but I was logged out of my account (like "properly" logged out -- not the "undefined" phantom account state as in #304). I went on to log in in the settings with my mail and password. after this, the app downloaded everything from the server again, such that I ended up with a duplicate of all notes and all tags. this then was also synced to my desktop client.

as you see in the Folder sidebar, notes are duplicated in a way that subfolders and their duplicates are listed under the first copy of the parent folder. one of my notes/folders (dark pink on the below screenshot) didn't duplicate but quadruplicate. I have absolutely no idea why, it wasn't even the note that I was editing when it crashed.

![image](https://user-images.githubusercontent.com/64873595/100393832-4c863680-303b-11eb-80db-c099cf4a6025.png)

in parallel I also had [all my extensions duplicated on desktop](https://github.com/standardnotes/desktop/issues/636), which may be because of this bug.

**Comment by JaspalSuri** [comment]
Hi @chaserene,

Thank you so much for all of the detailed reports that you've provided so far. I'm sorry for all of the bad luck that you've been having with the apps since the launch of v3.5. 🙁 

If you haven't already, have you cleared the app's storage and cache just in case? I know that the app has offered to clear any local data, but I just want to take note of this so that way I can add it to my report. (I apologize if you've mentioned this before.)

Regarding all the duplicate items, can you open the Batch Manager extension on the Desktop via the Extensions menu and see if it's able to detect duplicate **Notes**, **Components**, and **Tags**? Feel free to delete them if it does, but be careful with the tags by making sure that the duplicate tag isn't attached to any notes. If this isn't reflected in the Batch Manager's UI, please let me know.

**Comment by chaserene** [comment]
> If you haven't already, have you cleared the app's storage and cache just in case?

I have done that in the past, but I've gone through endless updates, crashes and logins/outs since the last time I've done that. I always cleared it through the option that the app offered, but manually, it's been a long time, so I guess the answer is no.

>  can you open the Batch Manager extension on the Desktop via the Extensions menu and see if it's able to detect duplicate

it did detect duplicate notes and I think it cleared them up successfully. there were no hits for components, since I already cleared those up manually a bit earlier. it also detected duplicate tags, but I didn't proceed with the deleting because what you said. do we know which duplicate will the Batch Manager delete? if it deletes the tag that is associated with the respective note, I'll have a lot of extra work re-assigning tags. I think I'll clear up the tags manually too.

**Comment by JaspalSuri** [comment]
> I have done that in the past, but I've gone through endless updates, crashes and logins/outs since the last time I've done that. I always cleared it through the option that the app offered, but manually, it's been a long time, so I guess the answer is no.

Thank you for letting me know. I'm sorry to hear that. Given that you've experienced several issues since and the app hasn't been very usable, it might not hurt to export backups of your notes from the mobile and web/desktop apps before clearing the app's storage and cache again. 

(Re: duplicate tags) Thank you for taking the time to do that. I'll make sure to put in a request for identifying unassigned tags for the dev team to review.
